### ISSUE TITLE
Disable bindings when typing in vscode terminal

### ISSUE THREAD

**Issue by youyuanwu** [tuttieee/vscode-emacs-mcx#344]
File opened in nano editor in vscode terminal needs to be closed by pressing Ctrl+x. But Ctrl+x always looks for the second input according to this key binding. Therefore the nano editor cannot be closed.

I encounter this issue because sometimes I do git merge manually and nano editor gets opened by git and I could not close it.

Maybe this extension should disable all key bindings when terminal is focused.

**Comment by tuttieee** [comment]
Good suggestion, let's fix it.
Could you do it? (This could be a hint: https://github.com/tuttieee/vscode-emacs-mcx/blob/master/DEVELOPMENT.md#keybindings-generation )
If not, I will.

**Comment by youyuanwu** [comment]
You mean restrict ctrl+x commands to non-terminal focus? 
I can give it a try. But how do I test this? when I run launch extension, the debug window don't seem to have the new key bindings. If I disable all extensions when lauch-extension in debug mode, I don't seem to have any key bindings.

**Comment by tuttieee** [comment]
> You mean restrict ctrl+x commands to non-terminal focus?

Yes. More precisely, I think adding `"when": "editorFocus"` to all keybindings starting with `C-x` may work (though I have not tried).

> when I run launch extension, the debug window don't seem to have the new key bindings.

mmm, that's strange. It should work... Is `package.json` updated before you launch the debug window?

**Comment by youyuanwu** [comment]
I was using WSL extension in vscode. I tried launch extension in native windows environment, the extension shows up as expected. This might be a bug/limitation in WSL extension. 

I will try add editorFocus and see the behavior.

**Comment by youyuanwu** [comment]
In windows environment I encounter this error when running unit-tests:
Three tests failed for the same reason.

```
  3) kill, yank, yank-pop
       with empty initial text
         yankPop does not work if selectAll is executed after previous yankPop:
     Error: command 'selectAll' not found
```

**Comment by tuttieee** [comment]
Thank you for trying and resolving.

The `selectAll` issue has started to show from VSCode 1.47.0 which was released last week.
I think `selectAll` command had been deprecated and removed at the release. I fixed the issue by #365 .

Anyway, your PR looks good and passes CI now. Thank you again.
